falcon/util/misc.py

```diff
@@ -190,7 +203,7 @@ def http_date_to_dt(http_date: str, obs_date: bool = False) -> datetime.datetime
     # Loop through the formats and return the first that matches
     for time_format in time_formats:
         try:
-            return _strptime(http_date, time_format)
+            return _strptime(http_date, time_format).replace(tzinfo=_UTC_TIMEZONE)
```

Comment:
This replacement doesn't seem to adhere to the now-obsolete [RFC 850, Section 2.1.4](https://datatracker.ietf.org/doc/html/rfc850#section-2.1.4) `TIMEZONE` formats.

Either we need to comply with the RFC, or deprecate/drop obs-date altogether.

Reply:
To be fair, OTOH, `datetime.datetime.strptime()` cannot parse the majority of 850's timezones like `EST` or `MDT` either. In that light, this change is probably acceptable, but I suggest we add a `NOTE` for posterity.

Reply:
Added a `NOTE` in #2308.